.github/main.workflow

```diff
@@ -0,0 +1,82 @@
+workflow "Continuous Integration" {
+  on = "push"
+  resolves = ["Bandit", "Black", "Pylint", "Test coverage"]
+}
+
+action "Bandit" {
+  uses = "./.github/actions/run"
+  args = [
+    "bandit -c .bandit.yml -r openscm"
+  ]
+  env = {
+    PYTHON_VERSION = "3.7"
+    PIP_PACKAGES = "bandit"
+  }
+}
+
+action "Black" {
+  uses = "./.github/actions/run"
+  args = [
+    "black --check openscm tests setup.py --exclude openscm/_version.py"
+  ]
+  env = {
+    PYTHON_VERSION = "3.7"
+    PIP_PACKAGES = "black"
+  }
+}
+
+action "Pylint" {
+  uses = "./.github/actions/run"
+  args = [
+    "pylint openscm"
+  ]
+  env = {
+    PYTHON_VERSION = "3.7"
+    PIP_PACKAGES = "pylint .[dev]"
+  }
+}
+
+action "Test coverage" {
+  uses = "./.github/actions/run"
+  args = [
+    "pytest -rfsxEX --cov=openscm tests --cov-report term-missing",
+    "pytest -rfsxEX --nbval ./notebooks --sanitize ./notebooks/tests_sanitize.cfg",
+    "if ! coverage report --fail-under=\"$MIN_COVERAGE\"",
+    "then",
+    "    echo",
+    "    echo \"Error: Coverage has to be at least ${MIN_COVERAGE}%\"",
+    "    exit 1",
+    "fi"
+  ]
+  env = {
+    PYTHON_VERSION = "3.7"
+    MIN_COVERAGE = "100"
```

Comment:
surely not 100?

Reply:
right now we are at 100. but we could also be more tolerant here.

Reply:
let's leave it for now, then when we want to reduce it we can discuss